Model/Ui/CapabilitiesConfigProvider.php

```diff
@@ -59,9 +63,43 @@ public function getConfig()
 
             // filter only active backends
             if (array_key_exists($code, $backends)) {
-                $configs['omise_payment_list'][$code]= $backends[$code];
+                if ($code === 'omise_offsite_shopeepay') {
+                    $configs['omise_payment_list'][$code] = $this->getShopeeBackendByType($backends[$code]);
+                } else {
+                    $configs['omise_payment_list'][$code]= $backends[$code];
+                }
             }
         }
+
         return $configs;
     }
+
+    /**
+     * Return the right ShopeePay backend depending on the platform and availability of
+     * the backend in the capability
+     */
+    private function getShopeeBackendByType($shopeeBackends)
+    {
+        $jumpAppBackend = [];
+        $mpmBackend = [];
+
+        // Since ShopeePay will have two types i.e shopeepay and shopeepay_jumpapp,
+        // we split and store the type in separate variables.
+        foreach ($shopeeBackends as $backend) {
+            if ($backend->type === Shopeepay::JUMPAPP_ID) {
+                $jumpAppBackend[] = $backend;
+            } else {
+                $mpmBackend[] = $backend;
+            }
+        }
+
+        $isShopeepayJumpAppEnabled = $this->capabilities->isBackendEnabled(Shopeepay::JUMPAPP_ID);
+        $isShopeepayEnabled = $this->capabilities->isBackendEnabled(Shopeepay::ID);
+
+        if ($this->helper->isMobilePlatform() && $isShopeepayJumpAppEnabled) {
+            return $jumpAppBackend;
+        }
+
+        return $isShopeepayEnabled ? $mpmBackend : $jumpAppBackend;
```

Comment:
here too i think maybe we want default to be shopeepay

Reply:
Here we return the backend, not the source. Although the logic is same as above. If the user is using mobile and jump app is enabled then we return shopeepay jump app backend. If not then we check if Shopeepay dierct is enabled or not. If yes we return direct backend else the jump app because if `shopeepay` is not enabled then `shopeepay_jumpap` must be enabled in order for this code to get executed. If none of them is enable then this code will never execute.